packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts

```diff
@@ -566,8 +568,103 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
     }
   }
 
+  const btcImportedFundsSweepEffectiveBalance = function* () {
+    try {
+      yield take(actionTypes.core.data.btc.FETCH_BTC_DATA_SUCCESS)
+      const addressesR = yield select(selectors.core.common.btc.getActiveAddresses)
+      const addresses = addressesR.getOrElse([]).filter(prop('priv'))
+      const hasEffectiveBalances: ImportedBtcAddressList = []
+      // eslint-disable-next-line no-restricted-syntax
+      for (const addr of addresses) {
+        if (addr.info.final_balance > 0) {
+          let payment = coreSagas.payment.btc.create({
+            network: networks.btc
+          })
+          payment = yield payment.init()
+          payment = yield payment.from(addr.addr, ADDRESS_TYPES.LEGACY)
+          const defaultFeePerByte = path(['fees', 'regular'], payment.value())
+
+          payment = yield payment.fee(defaultFeePerByte)
+          const effectiveBalance = prop('effectiveBalance', payment.value())
+          if (effectiveBalance > 0) {
+            hasEffectiveBalances.push({
+              address: addr.addr,
+              balance: addr.info.final_balance
+            })
+          }
+        }
+      }
+      yield put(A.btcImportedFundsSweepEffectiveBalanceSuccess(hasEffectiveBalances))
+    } catch (e) {
+      yield put(A.btcImportedFundsSweepEffectiveBalanceFailure([]))
+    }
+  }
+
+  const btcImportedFundsSweep = function* (action) {
+    const { payload } = action
+    yield put(A.btcImportedFundsSweepLoading())
+    try {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const addr of payload) {
+        const accounts = (yield select(selectors.core.common.btc.getAccountsBalances)).getOrElse([])
+        const defaultIndex = yield select(selectors.core.wallet.getDefaultAccountIndex)
+
+        const defaultAccount = accounts.filter((acc) => acc.index === defaultIndex)[0]
+
+        const receiveIndexMultiaddr = (yield select(
```

Comment:
I dont see this as blocking the release, but one improvement here would have been to yield the initial index once, and then just increment on that. Instead, we are yielding a fetch to multiaddr each time.